public/app/plugins/panel/barchart/BarChartPanel.tsx

```diff
@@ -23,13 +17,13 @@ export const BarChartPanel: React.FunctionComponent<Props> = ({
   fieldConfig,
   onFieldConfigChange,
 }) => {
-  if (!data || !data.series?.length) {
-    return (
-      <div className="panel-empty">
-        <p>No data found in response</p>
-      </div>
-    );
-  }
+  const orientation = useMemo(() => {
+    if (!options.orientation || options.orientation === VizOrientation.Auto) {
+      return width < height ? VizOrientation.Horizontal : VizOrientation.Vertical;
+    }
+
+    return options.orientation;
+  }, [width, height, options.orientation]);
```

Comment:
@leeoniya moved this calculation out of config builder. Feels like it's the panel's responsibility to define the viz orientation rather than visualization's.

Reply:
just depends on who we want to handle "auto" -- for the gauges, it is also in the vis

Reply:
Actually, it's in the VizRepeater which is outside the Visualization